docs/howto/upgrade_from_0_8_to_0_10.md

```diff
@@ -0,0 +1,227 @@
+    end
+
+    alias_method :options, :instance_options
+
+    # Since attributes could be read from the `object` via `method_missing`,
+    # the `try` method did not behave as before. This patches `try` with the
+    # original implementation plus the addition of
+    # ` || object.respond_to?(a.first)` to check if the object responded to
+    # the given method.
+    def try(*a, &b)
+      if a.empty? || respond_to?(a.first) || object.respond_to?(a.first)
+        try!(*a, &b)
+      end
+    end
+
+    # AMS 0.8 would return nil if the serializer was initialized with a nil
+    # resource.
+    def serializable_hash(adapter_options = nil,
+                          options = {},
+                          adapter_instance =
+                            self.class.serialization_adapter_instance)
+      object.nil? ? nil : super
+    end
+  end
+end
+
+```
+Add this class to your app however you see fit. This is the class that your existing serializers 
+that inherit from `ActiveMode::Serializer` should inherit from.
+
+### 3. Add `VersionEightCollectionSerializer`
+#### Code
+```ruby
+module ActiveModel
+  class Serializer
+    class VersionEightCollectionSerializer < CollectionSerializer
+      # In AMS 0.8, passing an ArraySerializer instance with a `root` option
+      # properly nested the serialized resources within the given root.
+      # Ex.
+      #
+      # class MyController < ActionController::Base
+      #   def index
+      #     render json: ActiveModel::Serializer::ArraySerializer
+      #       .new(resources, root: "resources")
+      #   end
+      # end
+      #
+      # Produced
+      #
+      # {
+      #   "resources": [
+      #     <serialized_resource>,
+      #     ...
+      #   ]
+      # }
+      def as_json(options = {})
+        if root
+          {
+            root => super
+          }
+        else
+          super
+        end
+      end
+
+      # AMS 0.8 used `DefaultSerializer` if it couldn't find a serializer for
+      # the given resource. When not using an adapter, this is not true in
+      # `0.10`
+      def serializer_from_resource(resource, serializer_context_class, options)
+        serializer_class = 
+          options.fetch(:serializer) { serializer_context_class.serializer_for(resource) }
+
+        if serializer_class.nil? # rubocop:disable Style/GuardClause
+          DefaultSerializer.new(resource, options)
+        else
+          serializer_class.new(resource, options.except(:serializer))
+        end
+      end
+
+      class DefaultSerializer
+        attr_reader :object, :options
+
+        def initialize(object, options={})
+          @object, @options = object, options
+        end
+
+        def serializable_hash
+          @object.as_json(@options)
+        end
+      end
+    end
+  end
+end
+```
+Add this class to your app however you see fit. This is the class that existing uses of 
+`ActiveMode::ArraySerializer` should be changed to use.
```

Comment:
do you mean this as a search/replace in source code? `git grep -l ArraySerializer | xargs perl -pi -w -e '/s/ActiveModel::ArraySerializer/ActiveModel::Serializer::VersionEightCollectionSerializer/g;'`

also, typo `s/ActiveMode/ActiveModel`